R/install-gitlab.R

```diff
@@ -70,12 +70,12 @@ gitlab_remote <- function(repo,
 remote_download.gitlab_remote <- function(x, quiet = FALSE) {
   dest <- tempfile(fileext = paste0(".tar.gz"))
 
-  src_root <- build_url(x$host, x$username, x$repo)
+  src_root <- build_url(x$host, x$username, x$repo, x$subdir)
   src <- paste0(src_root, "/repository/archive.tar.gz?ref=", utils::URLencode(x$ref, reserved = TRUE))
 
   if (!quiet) {
-    message("Downloading GitLab repo ", x$username, "/", x$repo, "@", x$ref,
-            "\nfrom URL ", src)
+    message("Downloading GitLab repo ", gsub("/$", "", x$username, "/", x$repo, "/", x$subdir),
```

Comment:
Are you actually getting a trailing slash in the subdirectory or just being cautious? In general I would consider specs with a trailing slash to be invalid input, and actually it seems `parse_git_repo()` strips them anyway, so you can remove the gsub here and below I think.

```r
> parse_git_repo("foo/bar/baz/")
$username
[1] "foo"

$repo
[1] "bar"

$subdir
[1] "baz"
```

Reply:
Oh now that I look at this again this handles the case when subdir is empty. I think a better approach is to use `paste0(c(x$username, x$repo, x$subdir), collapse = "/")`